docs/source/basics/chat_datasets.rst

```diff
@@ -0,0 +1,192 @@
+per sample. The list of messages could include a system prompt, multiple turns between user and assistant, and tool calls/returns.
+
+.. code-block:: text
+
+    |  conversations                         |
+    |----------------------------------------|
+    | [{"role": "user", "content": Q1},      |
+    |  {"role": "assistant", "content": A1}] |
+
+The conversation structure in the dataset will be reformatted to follow torchtune's :class:`~torchtune.data.Message` structure.
+The above will be converted to:
+
+.. code-block:: python
+
+    messages = [
+        Message(role="user", content="Q1"),
+        Message(role="assistant", content="A1"),
+    ]
+
+The list of messages is tokenized by the model tokenizer with the appropriate model-specific special tokens added
+(such as beginning-of-sequence, end-of-sequence, and others).
+
+.. code-block:: python
+
+    from torchtune.models.phi3 import phi3_mini_tokenizer
+
+    p_tokenizer = phi3_mini_tokenizer("/tmp/Phi-3-mini-4k-instruct/tokenizer.model")
+    tokens, mask = p_tokenizer.tokenize_messages(messages)
+    print(tokens)
+    # [1, 32010, 29871, 13, 29984, 29896, 32007, 29871, 13, 32001, 29871, 13, 29909, 29896, 32007, 29871, 13]
+    print(p_tokenizer.decode(tokens))
+    # '\nQ1 \n \nA1 \n'
+
+As an example, you can see the schema of the `SlimOrca dataset <https://huggingface.co/datasets/Open-Orca/SlimOrca-Dedup>`_.
+
+Specifying conversation style
+-----------------------------
+
+The structure of the conversation in the raw dataset can vary widely with different role names, different fields indicating
+where the message content name, and other ways. There are a few standardized formats that are common across many datasets.
+We have built-in converters to convert these standardized formats into a list of torchtune :class:`~torchtune.data.Message`
+that follows this format:
+
+.. code-block:: python
+
+    [
+        {
+            "role": "system" | "user" | "assistant" | "ipython",
+            "content": <message>,
+        },
+        ...
+    ]
+
+``"sharegpt"``
+^^^^^^^^^^^^^^
+The associated message transform is :class:`~torchtune.data.ShareGPTToMessages`. The expected format is:
+
+.. code-block:: python
+
+    {
+        "conversations": [
+            {
+                "from": "system" | "human" | "gpt",
+                "value": <message>,
+            },
+            ...
+        ]
+    }
+
+``"json"``
+^^^^^^^^^^
+The associated message transform is :class:`~torchtune.data.JSONToMessages`. The expected format is:
+
+.. code-block:: python
+
+    {
+        "messages": [
+            {
+                "role": "system" | "user" | "assistant",
+                "content": <message>,
+            },
+            ...
+        ]
+    }
+
+If your dataset does not fit one of the above conversation styles, then you will need to create a custom message transform.
+
+Loading chat datasets from Hugging Face
```

Comment:
This is a bit out of order from how you do things on the instruct dataset page, right? There you lead with loading the dataset which I think is nicer. I assume you're doing it here because `conversation_style` hasn't been defined yet. But another way to frame it is just "let's load a dataset.. what are all these different fields?" then use that to motivate the exposition of `conversation_style`. Personally I think that's better cause you give people a working code snippet right from the outset